src/modules/endpoint/sagas/watchFetchNews.js

```diff
@@ -0,0 +1,222 @@
+          : yield call(
+            loadTunews,
+            language,
+            FIRST_PAGE_INDEX,
+            TUNEWS_FETCH_COUNT_LIMIT
+          ))
+
+      const insert: PushNewsActionType = {
+        type: 'PUSH_NEWS',
+        params: {
+          news,
+          newsId,
+          cityLanguages: languages,
+          key,
+          language,
+          city,
+          type,
+          page: FIRST_PAGE_INDEX,
+          hasMoreNews: true
+        }
+      }
+      yield put(insert)
+    } else {
+      const allAvailableLanguages = !newsId ? new Map(languages.map(language => [language.code, null])) : new Map()
+
+      const failed: FetchNewsFailedActionType = {
+        type: 'FETCH_NEWS_FAILED',
+        params: {
+          message: 'Could not load news.',
+          code: ErrorCodes.PageNotFound,
+          allAvailableLanguages,
+          newsId: null,
+          key,
+          type,
+          language,
+          city
+        }
+      }
+      yield put(failed)
+    }
+  } catch (e) {
+    console.error(e)
+    const failed: FetchNewsFailedActionType = {
+      type: 'FETCH_NEWS_FAILED',
+      params: {
+        message: `Error in fetch news: ${e.message}`,
+        code: fromError(e),
+        key,
+        city,
+        language,
+        type,
+        newsId,
+        allAvailableLanguages: new Map()
+      }
+    }
+    yield put(failed)
+  }
+}
+
+export function * fetchMoreNews (
+  dataContainer: DataContainer,
+  action: FetchMoreNewsActionType
+): Saga<void> {
+  const {
+    city,
+    language,
+    newsId,
+    key,
+    criterion,
+    type,
+    page,
+    previouslyFetchedNews
+  } = action.params
+  try {
+    const peeking = yield select(state =>
+      isPeekingRoute(state, { routeCity: city })
+    )
+
+    const loadCriterion = new ContentLoadCriterion(criterion, peeking)
+
+    // Only get languages if we've loaded them, otherwise we're peeking
+    const cityLanguages = loadCriterion.shouldLoadLanguages()
+      ? yield call(dataContainer.getLanguages, city)
+      : []
+
+    const news = yield call(
+      loadTunews,
+      language,
+      page,
+      TUNEWS_FETCH_COUNT_LIMIT
+    )
+
+    const insert: PushNewsActionType = {
+      type: 'PUSH_NEWS',
+      params: {
+        news,
+        previouslyFetchedNews,
+        newsId,
+        hasMoreNews: news.length === TUNEWS_FETCH_COUNT_LIMIT,
+        cityLanguages,
+        key,
+        language,
+        city,
+        type,
+        page
+      }
+    }
+    yield put(insert)
+  } catch (e) {
+    console.error(e)
+    const failed: FetchNewsFailedActionType = {
+      type: 'FETCH_NEWS_FAILED',
+      params: {
+        message: `Error in fetch news: ${e.message}`,
+        code: fromError(e),
+        key,
+        city,
+        language,
+        type,
+        newsId,
+        allAvailableLanguages: new Map()
```

Comment:
@klinzo Regading making `allAvailableLanguages` as `new Map()` instead of `null` this make action  FETCH_NEWS_FAILED fails as it does this `   
case 'FETCH_NEWS_FAILED': {
        const { message, key, allAvailableLanguages, newsId, type, ...rest } = action.params
        return {
          ...state,
          newsRouteMapping: {
            ...state.newsRouteMapping,
            [key]: allAvailableLanguages
              ? { status: 'languageNotAvailable', type, allAvailableLanguages, ...rest }
              : { status: 'error', message, newsId, type, ...rest }
          }
        }
      }`
In other words, when network request fail bc of no internet/wifi, I got **language not available error** instead of **UnKnown error occurred**
IMO we should put it back to null to avoid clashing between two error types, what do you think?

Reply:
Yeah then put it back, thanks for testing :+1: